### ISSUE TITLE
Unable to add other grenades, and using !udm leaves you without weapons, even after 'guns'

### ISSUE THREAD

**Issue by GR-Code** [backraw/udm#7]
This is my first attempt at using SourcePython, so forgive any lapses in setup. I'm using the latest build #695, and fetched the files from your github rep instead of the release zip. Couple of problems : 

1. I can only use the HE grenade. Is there any way you can enable the other grenade types too?
2. After using !udm for test purposes, I found that I was stuck without weapons until I reconnected to the server. Using "guns" didn't help.

**Comment by thetredev** [comment]
Yes, you can only use HE grenades, that's on purpose. I can implement config options to use different grenades, too, if you wish. After using `!udm` you should get your weapons back when you close the admin menu using the `9` key. Didn't that work?

**Comment by GR-Code** [comment]
> I can implement config options to use different grenades, too, if you wish.

If you can, that'd be perfect.

>After using !udm you should get your weapons back when you close the admin menu using the 9 key.

Didn't work for me. Was weaponless even after closing the menu. Had to rejoin to get them back.

Also quick question, but is it possible to somehow enforce a map change after 1 round by default? Have a map that refuses to end unless we play 60 rounds, and even if I override the cvars it affects, the map overrides them at the end of every round.

**Comment by thetredev** [comment]
> Didn't work for me. Was weaponless even after closing the menu. Had to rejoin to get them back.

Oh! Which game are you trying it on? I thought I used it recently. I could be wrong, tho.

> Also quick question, but is it possible to somehow enforce a map change after 1 round by default? Have a map that refuses to end unless we play 60 rounds, and even if I override the cvars it affects, the map overrides them at the end of every round.

I'll definitely try that! Sounds like a good feature. I'm mostly free during the weekends, so development throughout the week could turn out a bit sparsely, just so you know.

**Comment by GR-Code** [comment]
Oh silly me, forgot to mention. CS:GO Dathost server, no extra SM plugins.

Let me know if I can help in any way. Looking forward to the addition of other grenade types soon.

**Comment by thetredev** [comment]
No problem at all. I can create a second branch for you to test the grenade functionality. Did you download the repo as a zip file or did you clone the git repo itself?

**Comment by GR-Code** [comment]
Zip file. Could clone it though.

**Comment by thetredev** [comment]
Please clone, then you can easily switch branches.

**Comment by GR-Code** [comment]
Noted. Do let me know when you're done.

**Comment by thetredev** [comment]
Let's discuss the grenade options further. What do you think of this concept?
```
// ----------------------------------
//    * HE Grenade Behavior
// ----------------------------------

// Options
//   * 0 = Off
//   * 1 = Equip on spawn
//   * 2 = Equip on spawn and on each HE grenade kill
//   * 3 = Equip on spawn and after each detonation
// Default Value: 2
// High Explosive grenade behavior
   udm_equip_hegrenade 2


// ----------------------------------
//    * Flashbang Grenade Behavior
// ----------------------------------

// Options
//   * 0 = Off
//   * 1 = Equip on spawn
//   * 2 = Equip on spawn and after each detonation
//   * 3 = Equip on spawn and after each kill caused by flashing an enemy
// Default Value: 3
// Flashbang grenade behavior
   udm_equip_flashbang 3


// ----------------------------------
//    * Smoke Grenade Behavior
// ----------------------------------

// Options
//   * 0 = Off
//   * 1 = Equip on spawn
//   * 2 = Equip on spawn and after each detonation
//   * 3 = Equip on spawn and after killing an enemy trough the smoke
// Default Value: 3
// Smoke grenade behavior
   udm_equip_smokegrenade 3


// ----------------------------------
//    * Molotov Behavior
// ----------------------------------

// Options
//   * 0 = Off
//   * 1 = Equip on spawn
//   * 2 = Equip on spawn and after each detonation
//   * 3 = Equip on spawn and after killing an enemy with the help of fire
//   * 4 = Equip on spawn and after killing an enemy by fire
// Default Value: 4
// Molotov behavior
   udm_equip_molotov 4


// ----------------------------------
//    * Incendiary Grenade Behavior
// ----------------------------------

// Options
//   * 0 = Off
//   * 1 = Equip on spawn
//   * 2 = Equip on spawn and after each detonation
//   * 3 = Equip on spawn and after killing an enemy with the help of fire
//   * 4 = Equip on spawn and after killing an enemy by fire
// Default Value: 4
// Incendiary grenade behavior
   udm_equip_incendiary 4


// ----------------------------------
//    * Decoy Grenade Behavior
// ----------------------------------

// Options
//   * 0 = Off
//   * 1 = Equip on spawn
//   * 2 = Equip on spawn and after each detonation
// Default Value: 1
// Decoy grenade behavior
   udm_equip_decoy 1
```

**Comment by GR-Code** [comment]
That looks perfect. Also for convenience sake, would you rather continue this conversation elsewhere? Am available on Discord, but I'm open to suggestions if you have some.

**Comment by thetredev** [comment]
Of course. I need to check my account details tho. PM me the channel. :D

**Comment by GR-Code** [comment]
Can't really PM you on here. Hit me up @GR™#0444 on Discord if you can.

**Comment by thetredev** [comment]
Will do.

**Comment by thetredev** [comment]
I can't find you there. Try to add CheebaDigga#2084 please.

**Comment by GR-Code** [comment]
Done.

**Comment by thetredev** [comment]
@GR-Code sorry for being absent for so long. Had different things to do... I sent a friend request over at Discord with my new username.
